src/uwp/Microsoft.Build.Net.CoreRuntimeTask/ResourceHandlingTask.cs

```diff
@@ -144,55 +174,81 @@ public override bool Execute()
+                        bool containsFrameworkResources = false;
+                        if (!_state.IsUpToDate(assemblyFilePath.ItemSpec, out containsFrameworkResources, out resWInfoList))
                         {
-                            if (containsResources)
-                                unprocessedAssemblyList.Add(assemblyFilePath);
-                            
-                            if (unprocessedAssemblyList.Count == 0)
-                                firstNonFrameworkAssembly = assemblyFilePath;
+                            resWInfoList = ExtractAssemblyResWList(assemblyFilePath.ItemSpec, out containsFrameworkResources);
+
+                            if (resWInfoList != null)
+                            {
+                                FileInfo fiAssembly = new FileInfo(assemblyFilePath.ItemSpec);
+                                _state.Save(assemblyFilePath.ItemSpec, fiAssembly.LastWriteTimeUtc, containsFrameworkResources, resWInfoList);
+                            }
                         }
-                        else
+
+                        if (resWInfoList != null)
                         {
-                            TaskItem newTaskItem = new TaskItem(reswPath);
-                            newTaskItem.SetMetadata("NeutralResourceLanguage","en-US");
-                            newTaskItem.SetMetadata("ResourceIndexName",Path.GetFileNameWithoutExtension(reswPath));
-                            reswList.Add(newTaskItem);
+                            foreach (ResWInfo reswInfo in resWInfoList)
+                            {
+                                TaskItem newTaskItem = new TaskItem(reswInfo.ResWPath);
+                                newTaskItem.SetMetadata("ResourceIndexName", reswInfo.ResourceIndexName);
+                                if (!String.IsNullOrEmpty(reswInfo.NeutralResourceLanguage))
+                                {
+                                    newTaskItem.SetMetadata("NeutralResourceLanguage", reswInfo.NeutralResourceLanguage);
+                                }
+
+                                if (containsFrameworkResources)
+                                {
+                                    frameworkReswList.Add(newTaskItem);
+                                }
+                                else
+                                {
+                                    newTaskItem.SetMetadata("OriginalItemSpec", reswInfo.ResWPath); // Original GenerateResource behavior creates this additional metadata item on processed non-framework assemblies
+                                    reswList.Add(newTaskItem);
+                                }
+                            }
                         }
 
                     }
 
-                    UnprocessedAssemblyList = unprocessedAssemblyList.ToArray();
+                    UnprocessedAssemblyList = unprocessedAssemblyList.ToArray(); // For now this list will always be empty
```

Comment:
>unprocessedAssemblyList [](start = 46, length = 23)

why we didn't keep the list of assemblies that not having resources here? would it be better to have IsUpToDate return this info so you don't have to process the assemblies that not having resources at all?

Reply:
We've always discarded assemblies that do not contain resources. Also, Inserting into this list would defeat the purpose of the change because items that are inserted into here will be passed to ResGen after this, which may cause ResGen to crash. For example, an RS3 C# UWP Library that has no resources